src/Converter.cc

```diff
@@ -229,13 +240,304 @@ void Converter::ConvertImpl(tinyxml2::XMLElement *_elem,
+    tinyxml2::XMLElement *newModel = doc->NewElement("model");
+    newModel->SetAttribute("name", newModelName.c_str());
+
+    if (FindNewModelElements(_elem, newModel, found + 2))
+    {
+      Unnest(newModel);
+      _elem->InsertEndChild(newModel);
+
+      // since newModel is inserted at the end, point back to the top element
+      nextElem = _elem->FirstChildElement();
+
+      if (!firstUnnestedModel)
+        firstUnnestedModel = newModel;
+      // std::cout << "newModel:\n" << PrintElement(newModel) << std::endl;
+    }
+  }
+}
+
+/////////////////////////////////////////////////
+bool Converter::FindNewModelElements(tinyxml2::XMLElement *_elem,
+                                    tinyxml2::XMLElement *_newModel,
+                                    const size_t &_newNameIdx)
+{
+  bool unnestedNewModel = false;
+  std::string newModelName = _newModel->Attribute("name");
+
+  tinyxml2::XMLElement *elem = _elem->FirstChildElement(), *nextElem = nullptr;
+  while (elem)
+  {
+    nextElem = elem->NextSiblingElement();
+
+    std::string elemName = elem->Name();
+    std::string elemAttrName;
+
+    if (elem->Attribute("name"))
+      elemAttrName = elem->Attribute("name");
+
+    if (elemAttrName.empty() ||
+        elemAttrName.compare(0, _newNameIdx - 2, newModelName) != 0 ||
+        (elemName != "frame" && elemName != "joint" &&
+        elemName != "link" && elemName != "model"))
+    {
+      if (elemName != "gripper")
+      {
+        elem = nextElem;
+        continue;
+      }
+    }
+
+    // Child attribute name w/ newModelName prefix stripped except for //gripper
+    std::string childAttrName;
+    if (elemName != "gripper")
+    {
+      childAttrName = elemAttrName.substr(_newNameIdx);
+      elem->SetAttribute("name", childAttrName.c_str());
+    }
+
+    // strip new model prefix from //pose/@relative_to
+    tinyxml2::XMLElement *poseElem = elem->FirstChildElement("pose");
+    if (poseElem != nullptr && poseElem->Attribute("relative_to"))
+    {
+      std::string poseRelTo = poseElem->Attribute("relative_to");
+
+      if (poseRelTo.compare(0, _newNameIdx - 2, newModelName) == 0)
+      {
+        poseRelTo = poseRelTo.substr(_newNameIdx);
+        poseElem->SetAttribute("relative_to", poseRelTo.c_str());
+      }
+    }
+
+    if (elemName == "frame")
+    {
+      std::string attachedTo;
+
+      if (elem->Attribute("attached_to"))
+      {
+        attachedTo = elem->Attribute("attached_to");
+
+        SDF_ASSERT(attachedTo.compare(0, _newNameIdx - 2, newModelName) == 0,
+          "Error: Frame attribute 'attached_to' does not start with " +
+          newModelName);
+
+        // strip new model prefix from attached_to
+        attachedTo = attachedTo.substr(_newNameIdx);
+        elem->SetAttribute("attached_to", attachedTo.c_str());
+      }
+
+      // remove frame if childAttrName == __model__
+      if (childAttrName == "__model__" && elem->Attribute("attached_to"))
```

Comment:
I think this if block can go inside the previous if block (`if (elem->Attribute("attached_to"))`)  with the `&& ...` removed.

Reply:
dce6828593004010ad58c44ab45f8d0b7bf462be